tsdb/engine.go

```diff
@@ -60,6 +74,21 @@ func NewEngine(path string, walPath string, options EngineOptions) (Engine, erro
 	// Only bolt-based backends are currently supported so open it and check the format.
 	var format string
 	if err := func() error {
+		// if it's a dir then it's a tsm1 engine
```

Comment:
Well -- 1 thing to note. If someone runs with tsm1, upgrades to a future release with support for, say, `tsm2`, and then downgrades, how will the system responds? Checking the magic number will allow graceful exit. Not checking it and assuming its `tsm1` may result in hard-to-understand crashes on initialization.